posts/2023-06-12-streams-in-scala.adoc

```diff
@@ -0,0 +1,225 @@
+= Streams in scala: An introductory guide
+zanio
+v1.0, 2023-06-12
+:title: Streams in Scala
+:imagesdir: ../media/2023-06-12-streams-in-scala
+:lang: en
+:tags: [streams, scala-streams, LazyList, fs2, akka-streams, scala, zio-streams, call-by-name, reactive-stream, back-pressure]
+
+== Outline
+ 1. <<Why Streams>>
+ 2. <<Scala Stream>>
+ 3. <<call-by-name>>
+ 4. <<lazylist-implementation>>
+ 5. <<stream-processing-libraries>>
+ 6. <<Conclusion>>
+ 7. <<References>>
+
+
+[[why-streams]]
+== Why Streams
+There are many reasons for using a stream-processing approach when writing software. In this blog post i'm going to focus on just one of those reasons: Memory Management.
+
+By processing elements one at a time streams enable you to avoid loading the entire dataset into memory, reducing the risk of encountering the dreaded 'Out of memory' error. Streams provide a lazy evaluation mechanism where elements are computed on-demand rather than being eagerly evaluated and stored in memory. This lazy evaluation, retaining only the necessary elements in memory, allows for efficient memory utilization especially when dealing with large datasets or potentially infinite sequences of data. With streams, you can confidently tackle memory-intensive tasks, knowing that the memory footprint is optimized, leading to more stable and scalable applications.
```

Comment:
```suggestion
By processing elements one at a time streams enables you to avoid loading the entire dataset into memory and reducing the risk of encountering the dreaded 'Out of memory' error. Streams provide a lazy evaluation mechanism where elements are computed on-demand rather than being eagerly evaluated and stored in memory. This lazy evaluation, retaining only the necessary elements in memory, allows for efficient memory utilization especially when dealing with large datasets or potentially infinite sequences of data. With streams, you can confidently tackle memory-intensive tasks, knowing that the memory footprint is optimized, leading to more stable and scalable applications.
```